### PR TITLE
Test for VLANs and bonds (LAGs)

### PR DESCRIPTION
**Pull request by milan-zededa**
Test to cover the functionality proposed and designed here:
https://wiki.lfedge.org/display/EVE/EVE+VLAN+support+-+create+VLAN+and+bond+interfaces

Note that VLANs and LAGs adds quite a bit of complexity into the configuration,
so it is rather easier to work with a custom device model than to try to add new CLI
commands with many attributes.

Signed-off-by: Milan Lenco <milan@zededa.com>

### CODE REVIEW

File: tests/network/testdata/vlans_and_bonds.txt [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+# as VMs-in-Containers for this test to pass.
+exec -t 2m bash check_vm_support.sh
+source .env
+[!env:with_hw_virt] skip 'Missing HW-assisted virtualization capability'
+
+# This test is based on ethernet loops, which are only supported on the QEMU dev model.
+{{$devmodel := EdenConfig "eve.devmodel"}}
+{{if not (eq $devmodel "ZedVirtual-4G")}}
+skip 'Test is supported only on QEMU'
+{{end}}
```
Comment (giggsoff) [inline tests/network/testdata/vlans_and_bonds.txt:44]: Unfortunately right now this combination (qemu+HW-assisted virtualization) is not handled by any CI workflows. Can we not use dhcp server in that test to make it a part of non-HW-accelerated (per PR) workflow?

**Review by rvs — APPROVED** [review]
LGTM

File: pkg/eden/qemu.go [hunk 2]
```diff
@@ -93,11 +98,24 @@ func StartEVEQemu(qemuARCH, qemuOS, eveImageFile, qemuSMBIOSSerial string, eveTe
...
 	}
-	qemuOptions += fmt.Sprintf(" -device %s,netdev=eth%d ", netDev, 1)
+	qemuOptions += fmt.Sprintf(" -device %s,netdev=eth%d ", netDev, ethIndex)
+	ethIndex++
 
 	if tapInterface != "" {
-		qemuOptions += fmt.Sprintf("-netdev tap,id=eth%d,ifname=%s", 2, tapInterface)
-		qemuOptions += fmt.Sprintf(" -device %s,netdev=eth%d ", netDev, 2)
+		qemuOptions += fmt.Sprintf("-netdev tap,id=eth%d,ifname=%s", ethIndex, tapInterface)
+		qemuOptions += fmt.Sprintf(" -device %s,netdev=eth%d ", netDev, ethIndex)
```
[the remark this answers is not in the document]
Comment (milan-zededa) [reply pkg/eden/qemu.go:106]: Good catch, fixed.

File: tests/network/testdata/vlans_and_bonds.txt (hunk 1 above)
Comment (milan-zededa) [reply tests/network/testdata/vlans_and_bonds.txt:44]: Unfortunately without Netdev capabilities inside applications I cannot think of any proper test for more advanced network features, such as VLAN and LAGs. And eventually we will have ROL where we will be able to run these tests. Testing applications as VMs inside containers (and not just containers) as part of CI should be our goal in any case.

**Pull request merged**